src/lib/browse/layers/areas/Pollution.svelte

```diff
@@ -0,0 +1,145 @@
+  // and QGIS, or http://www.extrium.co.uk/noiseviewer.html for the noise layer
+  $: info = {
+    NOx_viridis: ["22", "Data for 2022"],
+    PM25_viridis: ["21", "Data for 2022"],
+    PM10_viridis: ["22", "Data for 2022"],
+    NOxRoads_viridis: ["22", "Data for 2022"],
+    PM25Roads_viridis: ["14", "Data for 2022"],
+    PM10Roads_viridis: ["22", "Data for 2022"],
+    // TODO This one is disabled by leaving it out of the dropdown. The WMS
+    // server uses HTTP, not HTTPS, and doesn't work when deployed.
+    Noise: [
+      "NoiseE:RD_LQ16_R3",
+      "Annual average noise level for the 16-hour period between 0700-2300 (dB)",
+    ],
+  }[pollutant];
+
+  function wmsUrl(): string {
+    if (pollutant == "Noise") {
+      return `http://wms.extrium.co.uk/geoserver/NoiseE/wms`;
+    } else {
+      return `https://ukair.maps.rcdo.co.uk/ukairserver/services/aq_amb_2022/${pollutant}/MapServer/WMSServer`;
+    }
+  }
+
+  function title(pollutant: string): string {
+    return info![1];
+  }
+
+  function tilesUrl(pollutant: string): string {
+    let params = new URLSearchParams({
+      request: "GetMap",
+      version: "1.3.0",
+      format: "image/png",
+      crs: "EPSG:3857",
+      width: "256",
+      height: "256",
+      styles: "",
+      layers: info![0],
+    }).toString();
+    // Don't escape the {} in the bbox, so specify it manually below
+    return `${wmsUrl()}?bbox={bbox-epsg-3857}&${params}`;
+  }
+
+  function legendUrl(pollutant: string): string {
+    let params = new URLSearchParams({
+      request: "GetLegendGraphic",
+      version: "1.3.0",
+      format: "image/png",
+      layer: info![0],
+    }).toString();
+    return `${wmsUrl()}?${params}`;
+  }
+</script>
+
+<Checkbox id="pollution" bind:checked={show}>
+  Pollution
+  <span slot="right">
+    <HelpButton>
+      <p>
+        Most layers show air quality data from <ExternalLink
+          href="https://uk-air.defra.gov.uk/data/wms-services"
+        >
+          DEFRA
+        </ExternalLink>. The measurements are annual means, in units of
+        &micro;gm
+        <sup>3</sup>
+        . Note the particulate matter layers are not corrected for natural sources.
+      </p>
+      <p>
+        The noise layer is from <ExternalLink
+          href="http://www.extrium.co.uk/noiseviewer.html"
+        >
+          Extrium
+        </ExternalLink>, using 2017 data. See their <ExternalLink
+          href="http://www.extrium.co.uk/noiseviewer/FAQs.pdf"
+        >
+          FAQ
+        </ExternalLink>.
+      </p>
+      <OsOglLicense />
+    </HelpButton>
+  </span>
+</Checkbox>
+
+{#if show}
+  <Select
+    label="Pollutant"
+    id="pollutant"
+    choices={[
+      ["PM25_viridis", "Background PM2.5"],
+      ["PM10_viridis", "Background PM10"],
+      ["NOx_viridis", "Background NOx"],
+      ["PM25Roads_viridis", "Roadside PM2.5"],
+      ["PM10Roads_viridis", "Roadside PM10"],
+      ["NOxRoads_viridis", "Roadside NOx"],
```

Comment:
Does this need `["Noise", "Noises],` adding?

Reply:
Note this PR contains an attempt at the noise layer from http://www.extrium.co.uk/noiseviewer.html. It's disabled for two reasons:

    I'm not sure they mean for it to be used publicly, even though the WMS server happily serves content without CORS problems.
    They only have an HTTP endpoint. It works with npm run dev, but not on GH pages or GCS because of https://developer.mozilla.org/en-US/docs/Web/Security/Mixed_content.